### ISSUE TITLE
AttributeError: module 'optuna.integration.lightgbm' has no attribute 'train'

### ISSUE THREAD

**Issue by harupy** [optuna/optuna#817]
When building the doc, the following error occurs.

<img width="1056" alt="Screen Shot 2019-12-27 at 10 47 47" src="https://user-images.githubusercontent.com/17039389/71496378-600ea780-2896-11ea-829f-d64c9f22fca4.png">

https://circleci.com/gh/optuna/optuna/24947#tests/containers/0

**Comment by harupy** [comment]
The cause is that `lightgbm` is not installed in the doc build environment and the attribute `train` isn't attached to `optuna.integration.lightgbm`.

https://github.com/optuna/optuna/blob/683b3ca530a52154b191d3e7e19e1cee2a8ab82f/optuna/integration/lightgbm.py#L5-L34


Adding `lightgbm` to the document section in `setup.py` should fix this issue.

```diff
        'document': [
+           'lightgbm'
            'sphinx',
            'sphinx_rtd_theme',
        ],
```

https://github.com/optuna/optuna/blob/master/setup.py#L73-L75

**Comment by hvy** [comment]
Thanks for the report and the follow-up investigation. The behavior you describe makes sense. It should be okay to add the `lightgbm` requirement. Would you like to create a PR or should I?

**Comment by hvy** [comment]
We should actually treat these warnings as CI failures, let me raise a separate ticket for that.

**Comment by hvy** [comment]
Closing as it's fixed by #819.
